fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/CanonicalValue.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.util;
+
+/**
+ * A canonical value, with processing applied to identify the uri, version, and fragment
+ */
+public class CanonicalValue {
+
+    private final String uri;
+    private final String version;
+    private final String fragment;
+
+    /**
+     * @param uri
+     *          The uri portion of the canonical value
+     * @param version
+     *          The version portion of the canonical value, can be null
+     * @param fragment
+     *          The fragment portion of the canonical value, can be null
+     */
+    public CanonicalValue(String uri, String version, String fragment) {
+        this.uri = uri;
+        this.version = version;
+        this.fragment = fragment;
+    }
+
+    /**
+     * @return the uri
+     */
+    public String getUri() {
+        return uri;
+    }
+
+    /**
+     * @return the version
+     */
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * @return the fragment
+     */
+    public String getFragment() {
+        return fragment;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((fragment == null) ? 0 : fragment.hashCode());
```

Comment:
Objects.hashCode might be more convenient here

Reply:
changed to use Objects.hash(fragment, uri, version)